packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -34,35 +35,73 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
         }
 
         const { uid } = sessionToken;
-        const { recoveryKeyId, recoveryData } = request.payload;
+        const { recoveryKeyId, recoveryData, enabled } = request.payload;
 
-        await db.createRecoveryKey(uid, recoveryKeyId, recoveryData);
+        await db.createRecoveryKey(uid, recoveryKeyId, recoveryData, enabled);
 
         log.info('account.recoveryKey.created', { uid });
 
-        await request.emitMetricsEvent('recoveryKey.created', { uid });
+        if (enabled) {
+          await request.emitMetricsEvent('recoveryKey.created', { uid });
+
+          const account = await db.account(uid);
+
+          const { acceptLanguage, clientAddress: ip, geo, ua } = request.app;
+          const emailOptions = {
+            acceptLanguage,
+            ip,
+            location: geo.location,
+            timeZone: geo.timeZone,
+            uaBrowser: ua.browser,
+            uaBrowserVersion: ua.browserVersion,
+            uaOS: ua.os,
+            uaOSVersion: ua.osVersion,
+            uaDeviceType: ua.deviceType,
+            uid,
+          };
+
+          await mailer.sendPostAddAccountRecoveryEmail(
+            account.emails,
+            account,
+            emailOptions
+          );
+        }
 
-        const account = await db.account(uid);
+        return {};
+      },
+    },
+    {
+      method: 'POST',
+      path: '/recoveryKey/verify',
+      options: {
+        auth: {
+          strategy: 'sessionToken',
+        },
+        validate: {
+          payload: {
+            recoveryKeyId: validators.recoveryKeyId,
+          },
+        },
+      },
+      handler: async function(request) {
+        log.begin('verifyRecoveryKey', request);
 
-        const { acceptLanguage, clientAddress: ip, geo, ua } = request.app;
-        const emailOptions = {
-          acceptLanguage,
-          ip,
-          location: geo.location,
-          timeZone: geo.timeZone,
-          uaBrowser: ua.browser,
-          uaBrowserVersion: ua.browserVersion,
-          uaOS: ua.os,
-          uaOSVersion: ua.osVersion,
-          uaDeviceType: ua.deviceType,
-          uid,
-        };
+        const sessionToken = request.auth.credentials;
 
-        await mailer.sendPostAddAccountRecoveryEmail(
-          account.emails,
-          account,
-          emailOptions
-        );
+        if (sessionToken.tokenVerificationId) {
+          throw errors.unverifiedSession();
+        }
+
+        const { uid } = sessionToken;
+        const { recoveryKeyId } = request.payload;
+
+        // Attempt to retrieve a recovery key, if it exists and is not already enabled,
+        // then we enable it.
+        const recoveryKeyData = await db.getRecoveryKey(uid, recoveryKeyId);
```

Comment:
A quick drive-by to say, it looks like this route would let you test whether a given `recoveryKeyId` is valid, so should probably be subject to the same rate-limiting as other routes like `GET /recoveryKey/{recoveryKeyId}`.

Reply:
Ah yea that makes sense, I will make those changes. In meantime I am going to merge these changes into https://github.com/mozilla/fxa/pull/3957.